### PR TITLE
Automatically allocate ports for functional tests

### LINKED ISSUE #358: Flaky Test: Localization_ResourcesInFolder_ReturnLocalizedValue_Windows
```
[Microsoft.AspNetCore.Localization.FunctionalTests.LocalizationTest.Localization_ResourcesInFolder_ReturnLocalizedValue_Windows]
System.Exception : Command exited unexpectedly with exit code: -532462766
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Testing.TaskExtensions.<TimeoutAfter>d__0`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer.<StartSelfHostAsync>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer.<DeployAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Localization.FunctionalTests.TestRunner.<RunTestAndGetResponse>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Localization.FunctionalTests.TestRunner.<RunTestAndVerifyResponse>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

## Repros
4/25/17: http://aspnetci/viewLog.html?buildId=208285&buildTypeId=Lite_MvcTest&tab=buildLog&filter=important

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #358. This is the same thing we do [here](https://github.com/aspnet/Performance/blob/fa204e1b1878b2214a1f34bee29d0f00af41186a/src/Benchmarks.Utility/Helpers/IISTestManager.cs#L44).

### CODE REVIEW

File: test/Microsoft.AspNetCore.Localization.FunctionalTests/TestRunner.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@ namespace Microsoft.AspNetCore.Localization.FunctionalTests
 {
     public class TestRunner
     {
+        private const string ApplicationBasePath = "http://localhost:0";
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Localization.FunctionalTests/TestRunner.cs]: I think you should merge this to unflake the build but looking at the history of this comment https://github.com/aspnet/Performance/blob/fa204e1b1878b2214a1f34bee29d0f00af41186a/src/Benchmarks.Utility/Helpers/IISTestManager.cs#L43

It sounds like this is a temporary workaround that was never de-temporaried - is there another item that tracks this?

Comment (rynowak) [inline test/Microsoft.AspNetCore.Localization.FunctionalTests/TestRunner.cs]: It also looks like this code was copied from another repo and that's not a good solution for something that's going to live this long.

Comment (rynowak) [inline test/Microsoft.AspNetCore.Localization.FunctionalTests/TestRunner.cs]: Let's try to get to the bottom of this tomorrow and make sure we have a plan here.

**Comment by ryanbrandenburg** [comment]
Closing in favor of #361.

**Pull request closed without merging**